components/odh-notebook-controller/controllers/notebook_controller_test.go

```diff
@@ -1080,6 +1084,269 @@ var _ = Describe("The Openshift Notebook controller", func() {
+						APIServer: dspav1.ComponentDetailStatus{
+							ExternalUrl: "https://pipeline-api.example.com",
+						},
+					},
+				},
+			}
+			Expect(cli.Create(ctx, dspaObj)).To(Succeed())
+
+			By("Creating a COS3 credentials Secret")
+			s3CredSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: Namespace,
+				},
+				Data: map[string][]byte{
+					accessKeyKey: []byte("testaccesskey"),
+					secretKeyKey: []byte("testsecretkey"),
+				},
+			}
+			Expect(cli.Create(ctx, s3CredSecret)).To(Succeed())
+
+			By("Creating a Dashboard CR")
+			dashboard = &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "components.platform.opendatahub.io/v1alpha1",
+					"kind":       "Dashboard",
+					"metadata": map[string]interface{}{
+						"name":      dashboardInstanceName,
+						"namespace": Namespace,
+					},
+				},
+			}
+			dashboard.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "components.platform.opendatahub.io",
+				Version: "v1alpha1",
+				Kind:    "Dashboard",
+			})
+			Expect(cli.Create(ctx, dashboard)).To(Succeed())
+
+			By("Patching the Dashboard status.url and Ready condition to simulate readiness")
+			// Fetch the Dashboard to get a valid resourceVersion
+			Expect(cli.Get(ctx, client.ObjectKeyFromObject(dashboard), dashboard)).To(Succeed())
+			// Create a patch with updated status.url and conditions
+			dashboardPatch := dashboard.DeepCopy()
+			err := unstructured.SetNestedField(dashboardPatch.Object, "https://dashboard.example.com", "status", "url")
+			Expect(err).ToNot(HaveOccurred())
+			readyCondition := map[string]interface{}{
+				"type":   "Ready",
+				"status": "True",
+				"reason": "ComponentsAvailable",
+			}
+			err = unstructured.SetNestedSlice(dashboardPatch.Object, []interface{}{readyCondition}, "status", "conditions")
+			Expect(err).ToNot(HaveOccurred())
+			patch := client.MergeFrom(dashboard)
+			Expect(cli.Status().Patch(ctx, dashboardPatch, patch)).To(Succeed())
+
+			By("Waiting the Dashboard object is present and Ready")
+			time.Sleep(10 * time.Millisecond)
+
+			By("Creating Notebook")
+			notebook := createNotebook(notebookName, Namespace)
+			Expect(cli.Create(ctx, notebook)).To(Succeed())
+
+			By("Waiting for ds-pipeline-config Secret to be created")
+			Eventually(func() error {
+				return cli.Get(ctx, types.NamespacedName{Name: dsSecretName, Namespace: Namespace}, &corev1.Secret{})
+			}, 30*time.Second, 2*time.Second).Should(Succeed())
+
+			// This is needed until RHOAIENG-24545 bug get fix, related with the maybeRestartRunningNotebook func
+			By("Stopping the Notebook to allow volume injection")
+			stopPatch := notebook.DeepCopy()
+			if stopPatch.Annotations == nil {
+				stopPatch.Annotations = map[string]string{}
+			}
+			stopPatch.Annotations["kubeflow-resource-stopped"] = "true"
+			Expect(cli.Patch(ctx, stopPatch, client.MergeFrom(notebook))).To(Succeed())
```

Comment:
> Without the manual annotation trigger, the test fails with the webhook log clearly showing:
>
> Update blocked, notebook pod template would be changed by the webhook   {"notebook": "dspa-notebook", "namespace": "dspa-test-namespace", "diff": "{v1.PodSpec}.Volumes: [...] != []"}
> 
> To resolve this, I updated the test to set kubeflow-resource-stopped=true, which is fully supported by the webhook and reflects how users would safely trigger updates in production.

I was not complaining about the test code, I am complaining about production code! Until RHOAIENG-24545 bug gets fixed, this elyra secret will not work right for the first started notebook. And this is a regression from how it works in RHOAI 2.21.

We can't ship a regression, can we?

Reply:
We could tell our users to apply the workaround, or restart the first workbench they've started, but that does not change the fact we're introducing regression.

Reply:
The test code change is actually good, it simulates the user facing workaround of restarting workbench, so that's nice! But still, we're introducing a regression in this PR.

Reply:
In a real cluster env i don't have this issue tho (wrt updating etc) not blocking anything there. The issue appears on the test framework

Reply:
double checked, not working for me, sent you message on slack